.buildkite/gen-pipeline.sh

```diff
@@ -298,7 +285,7 @@ run_spark_integration() {
   # Horovod Spark Estimator tests
   if [[ ${test} != *"tf1_1_0"* && ${test} != *"tf1_6_0"* && ${test} != *"torch0_"* && ${test} != *"mpich"* && ${test} != *"oneccl"* ]]; then
     if [[ ${test} != *"tf2"* && ${test} != *"tfhead"* ]]; then
-      if [[ ! (  ${test} == *"py2"* && ${test} == *"gloo"* && ${test} != *"openmpi-gloo"* ) ]]; then
+      if [[ ! (  ${test} == *"gloo"* && ${test} != *"openmpi-gloo"* ) ]]; then
```

Comment:
now that there are only two conditions left, the overall negation should be applied to simplify the expression:

    ${test} != *"gloo"* || ${test} == *"openmpi-gloo"*